services/cirrus/cirrus.service.js

```diff
@@ -0,0 +1,83 @@
+'use strict'
+
+const Joi = require('@hapi/joi')
+const { isBuildStatus, renderBuildStatusBadge } = require('../build-status')
+const { BaseJsonService } = require('..')
+
+const schema = Joi.object({
+  status: Joi.alternatives()
+    .try(isBuildStatus, Joi.equal('unknown'))
+    .required(),
+}).required()
+const queryParamSchema = Joi.object({
+  task: Joi.string(),
+  script: Joi.string(),
+}).required()
+
+module.exports = class Cirrus extends BaseJsonService {
+  static get category() {
+    return 'build'
+  }
+
+  static get route() {
+    return {
+      base: 'cirrus',
+      pattern: 'github/:user/:repo/:branch*',
+      queryParamSchema,
+    }
+  }
+
+  static get examples() {
+    return [
+      {
+        title: 'Cirrus CI - Base Branch Build Status',
+        namedParams: { user: 'flutter', repo: 'flutter' },
+        pattern: '/github/:user/:repo',
+        queryParams: { task: 'analyze', script: 'test' },
+        staticPreview: this.render({ subject: 'Ci', status: 'passing' }),
+      },
+      {
+        title: 'Cirrus CI - Specific Branch Build Status',
+        pattern: '/github/:user/:repo/:branch',
+        namedParams: { user: 'flutter', repo: 'flutter', branch: 'master' },
+        queryParams: { task: 'analyze', script: 'test' },
+        staticPreview: this.render({ subject: 'Ci', status: 'passing' }),
+      },
+      {
+        title: 'Cirrus CI - Specific Task Build Status',
+        pattern: '/github/:user/:repo',
+        queryParams: { task: 'analyze' },
+        namedParams: { user: 'flutter', repo: 'flutter' },
+        staticPreview: this.render({ subject: 'analyze', status: 'passing' }),
+      },
+      {
+        title: 'Cirrus CI - Task and Script Build Status',
+        pattern: '/github/:user/:repo',
+        queryParams: { task: 'analyze', script: 'test' },
+        namedParams: {
+          user: 'flutter',
+          repo: 'flutter',
+        },
+        staticPreview: this.render({ subject: 'test', status: 'passing' }),
+      },
+    ]
+  }
+
+  static get defaultBadgeData() {
+    return { label: 'build' }
+  }
+
+  static render({ subject, status }) {
+    return renderBuildStatusBadge({ label: subject, status })
+  }
+
+  async handle({ user, repo, branch }, { script, task }) {
+    const json = await this._requestJson({
+      schema,
+      url: `https://api.cirrus-ci.com/github/${user}/${repo}.json`,
+      options: { qs: { branch, script, task } },
+    })
+
+    return this.constructor.render({ status: json.status })
```

Comment:
@calebcartwright you are right, `subject` of the json response will contain either `task` or `script` if any of them are provided. Then the code will be even simplier:

```suggestion
    return this.constructor.render(json)
```

Reply:
Exactly what I was thinking. That way we can defer that logic of determining the correct value of `subject` to Cirrus (and if things ever change on the Cirrus side our badges will automatically reflect that too).

Note that the `schema` will also have to be updated above to bring in the `subject` field, and I've added an inline suggestion above to do that. 

I'd still be inclined to replace the `Ci` subject with the standard `build` value for the badge messages to be consistent with our other build status badges. If you guys are 👍 with that then let's make sure that's handled in the `render` function

Reply:
Will update in on server side. Thanks for the suggestion. 🙌

Reply:
@RDIL  when you get a chance can you commit this suggestion? Everything else looks good to go so once this last piece is updated we should be ready to merge 👍